obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+        the S-file will be taken from this file if evtime is not set.
+    :type evtype: str
+    :param evtype: Event type letter code, e.g. L, R, D
+    :type userid: str
+    :param userid: 4-character SEISAN USER ID
+    :type outdir: str
+    :param outdir: Location to write S-file
+    :type overwrite: bool
+    :param overwrite: Overwrite an existing S-file, default=False
+    :type evtime: obspy.core.utcdatetime.UTCDateTime
+    :param evtime: If given this will set the timing of the S-file
+
+    :returns: str, S-file name
+    """
+
+    from obspy import read as obsread
+    import os
+    import datetime
+
+    if not evtime:
+        try:
+            st = obsread(wavefile)
+            evtime = st[0].stats.starttime
+        except:
+            raise IOError('Wavefile: ' + wavefile +
+                          ' is invalid, try again with real data.')
+    # Check that user ID is the correct length
+    if len(userid) != 4:
+        raise IOError('User ID must be 4 characters long')
+    # Check that outdir exists
+    if not os.path.isdir(outdir):
+        raise IOError('Out path does not exist, I will not create this: ' +
+                      outdir)
+    # Check that evtype is one of L,R,D
+    if evtype not in ['L', 'R', 'D']:
+        raise IOError('Event type must be either L, R or D')
+
+    # Generate s-file name in the format dd-hhmm-ss[L,R,D].Syyyymm
+    sfile = outdir + '/' + str(evtime.day).zfill(2) + '-' +\
+        str(evtime.hour).zfill(2) +\
+        str(evtime.minute).zfill(2) + '-' +\
+        str(evtime.second).zfill(2) + evtype + '.S' +\
+        str(evtime.year) +\
+        str(evtime.month).zfill(2)
+    # Check is sfile exists
+    if os.path.isfile(sfile) and not overwrite:
+        print('Desired sfile: ' + sfile + ' exists, will not overwrite')
+        for i in range(1, 10):
+            sfile = outdir + '/' + str(evtime.day).zfill(2) + '-' +\
+                str(evtime.hour).zfill(2) +\
+                str(evtime.minute).zfill(2) + '-' +\
+                str(evtime.second + i).zfill(2) + evtype + '.S' +\
+                str(evtime.year) +\
+                str(evtime.month).zfill(2)
+            if not os.path.isfile(sfile):
+                break
+        else:
+            msg = 'Tried generated files up to 20s in advance and found ' +\
+                'all exist, you need to clean your stuff up!'
+            raise IOError(msg)
+        # sys.exit()
+    f = open(sfile, 'w')
+    # Write line 1 of s-file
+    f.write(str(' ' + str(evtime.year) + ' ' +
+                str(evtime.month).rjust(2) +
+                str(evtime.day).rjust(2) + ' ' +
+                str(evtime.hour).rjust(2) +
+                str(evtime.minute).rjust(2) + ' ' +
+                str(float(evtime.second)).rjust(4) + ' ' +
+                evtype + '1'.rjust(58) + '\n'))
+    # Write line 2 of s-file
+    f.write(str(' ACTION:ARG ' + str(datetime.datetime.now().year)[2:4] + '-' +
```

Comment:
You should call `datetime.datetime.now()` once, or there is a very small chance that you will get two different "now"s which will only show up once in a blue moon exactly while running a 30-hour script that you need to have finished running yesterday.